app/src/main/java/org/mozilla/focus/telemetry/FirebaseEvent.java

```diff
@@ -0,0 +1,106 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 20; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.telemetry;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import org.mozilla.focus.utils.AppConstants;
+import org.mozilla.focus.utils.FirebaseHelper;
+
+class FirebaseEvent {
+
+    // limitation for event:
+    // Event names can be up to 40 characters long, may only contain alphanumeric characters and
+    // underscores ("_"), and must start with an alphabetic character. The "firebase_", "google_"
+    // and "ga_" prefixes are reserved and should not be used.
+    // see: https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.Event
+    static final int MAX_LENGTH_EVENT_NAME = 40;
+    static final String EVENT_NAME_SEPARATOR = "__";
+
+    // limitation fo param
+    // You can associate up to 25 unique Params with each Event type. Param names can be up to 40
+    // characters long, may only contain alphanumeric characters and underscores ("_"),
+    // and must start with an alphabetic character. Param values can be up to 100 characters long.
+    // The "firebase_", "google_" and "ga_" prefixes are reserved and should not be used.
+    // see: https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.Param
+    static final int MAX_PARAM_SIZE = 25;
+    static final int MAX_LENGTH_PARAM_NAME = 40;
+    static final int MAX_LENGTH_PARAM_VALUE = 100;
+
+    private String eventName;
+    private Bundle eventParam;
+
+
+    private FirebaseEvent(@NonNull String category, @NonNull String method, @Nullable String object, @Nullable String value) {
+
+        // append null string will append 'null'
+        // Android Studio also said we don't need to use a String builder here.
+        this.eventName = method + EVENT_NAME_SEPARATOR + object + EVENT_NAME_SEPARATOR + value;
+        if (!AppConstants.isReleaseBuild() && this.eventName.length() > MAX_LENGTH_EVENT_NAME) {
```

Comment:
What happens when we're in production, and the length is too long? The Event is still set and will be queued. (Will Firebase trim it? Or some exception or ignored? Which is our expected behavior?)

Reply:
We will be notified for invalid telemetries? Sounds good enough to me. Just to make sure there is no hazard and no unexpected behaviors.